### ISSUE TITLE
Cannot build OpenBLAS 0.3.20 with Fujitsu compiler

### ISSUE THREAD

**Issue by giordano** [xianyi/OpenBLAS#3692]
Trying to build OpenBLAS 3.2.0 with Fujitsu compiler on Fugaku with the following spec: `openblas@0.3.20%fj@4.8.0+ilp64 symbol_suffix=64_` (`+ilp64 symbol_suffix=64_` is irrelevant for this) but I get into the following error:
```
"../kernel/arm64/../generic/../simd/intrin.h", line 56: catastrophic error: cannot open source file "arm_neon.h"
  #include <arm_neon.h>
                       ^

1 catastrophic error detected in the compilation of "../kernel/arm64/../generic/dot.c".
Compilation terminated.
```
As far as I can see, there is indeed no `arm_neon.h`, only `arm_sve.h`.

**Comment by giordano** [comment]
Not using Spack (the error above may be a [Spack-specific issue](https://github.com/spack/spack/issues/31675)), I still get other errors.  Excerpt:
```console
$ make CC=fcc FC=frt INTERFACE64=1 SYMBOLSUFFIX=64_ LDFLAGS='-lfj90i -lfj90f -lfjsrcinfo -lelf' 'RANLIB=ranlib' 'libs' 'netlib' 'shared'
"level3_syrk_threaded.c", line 71: error: identifier "_Atomic" is undefined
  _Atomic
  ^

"level3_syrk_threaded.c", line 75: error: expected a ";"
     BLASLONG working[MAX_CPU_NUMBER][CACHE_LINE_SIZE * DIVIDE_RATE];
              ^

"level3_syrk_threaded.c", line 272: error: struct "<unnamed>" has no field "working"
  	while (job[mypos].working[i][CACHE_LINE_SIZE * bufferside]) {YIELDING;};
  	                  ^

"level3_syrk_threaded.c", line 345: error: struct "<unnamed>" has no field "working"
  	job[mypos].working[i][CACHE_LINE_SIZE * bufferside] = (BLASLONG)buffer[bufferside];
  	           ^

"level3_syrk_threaded.c", line 366: error: struct "<unnamed>" has no field "working"
  	  while(job[current].working[mypos][CACHE_LINE_SIZE * bufferside] == 0) {YIELDING;};
  	                     ^

"level3_syrk_threaded.c", line 380: error: struct "<unnamed>" has no field "working"
  			   sa, (FLOAT *)job[current].working[mypos][CACHE_LINE_SIZE * bufferside],
  			                             ^
 in expansion of macro "KERNEL_OPERATION" at line 379
  	  KERNEL_OPERATION(min_i, MIN(range_n[current + 1]  - xxx,  div_n), min_l, alpha,
  	  ^

"level3_syrk_threaded.c", line 392: error: struct "<unnamed>" has no field "working"
  	    job[current].working[mypos][CACHE_LINE_SIZE * bufferside] &= 0;
  	                 ^

"level3_syrk_threaded.c", line 437: error: struct "<unnamed>" has no field "working"
  			   sa, (FLOAT *)job[current].working[mypos][CACHE_LINE_SIZE * bufferside],
  			                             ^
 in expansion of macro "KERNEL_OPERATION" at line 436
  	  KERNEL_OPERATION(min_i, MIN(range_n[current + 1] - xxx, div_n), min_l, alpha,
  	  ^

"level3_syrk_threaded.c", line 452: error: struct "<unnamed>" has no field "working"
  	    job[current].working[mypos][CACHE_LINE_SIZE * bufferside] &= 0;
  	                 ^

"level3_syrk_threaded.c", line 474: error: struct "<unnamed>" has no field "working"
  	while (job[mypos].working[i][CACHE_LINE_SIZE * xxx] ) {YIELDING;};
  	                  ^

"level3_syrk_threaded.c", line 690: error: struct "<unnamed>" has no field "working"
  	  job[j].working[i][CACHE_LINE_SIZE * k] = 0;
  	         ^
```
[`build.log`](https://github.com/xianyi/OpenBLAS/files/9158795/build.log) (after a first pass which successfully built other kernels).


I wonder if Fujtisu compiler lies about its C11 compliance:
```console
$ fcc -dM -E - < /dev/null | grep __STDC_VERSION__
#define __STDC_VERSION__ 201112L
```

**Comment by martin-frbg** [comment]
Yes, that's what I'm wondering too - the c_check script already tries to make sure that at least stdatomic.h is present when the compiler claims compliance, but it currently does not check if that header actually defines _Atomic. (Though I guess it would be pretty wild if it did not - could you check that please ?).

**Comment by martin-frbg** [comment]
BTW you will want to build with TARGET=A64FX on Fugaku - unfortunately the addition of recent Neoverse cpus in 0.3.20 broke the sequence of an indexed list of cpu names in the autodetection code as an unnoticed side effect :(

**Comment by giordano** [comment]
```console
$ echo -e '#include <stdatomic.h>\nint main(void){}\n' | fcc -x c -
Line 1: catastrophic error: cannot open source file "stdatomic.h"
  #include <stdatomic.h>
                        ^

1 catastrophic error detected in this compilation.
Compilation terminated.
```

**Comment by martin-frbg** [comment]
Oops. Then how can HAVE_C11 be defined in the generated Makefile.conf ?

**Comment by martin-frbg** [comment]
Err, could you try compiling the current `develop` branch instead ? I have a feeling that is another thing already fixed (apart from c_check being completely replaced by a shell script)

**Comment by martin-frbg** [comment]
Right - #3579 for the malfunctioning c11 check and #3586 for the cpuname carnage. 0.3.21 should have been out a month ago but real life issues keep piling up and I'm long back to (supposedly) working on completely different things during business hours

**Comment by giordano** [comment]
I've got small progress trying to build the `develop` branch, with `TARGET=A64FX`:
```console
$ make CC=fcc FC=frt INTERFACE64=1 SYMBOLSUFFIX=64_ TARGET=A64FX LDFLAGS='-lfj90i -lfj90f -lfjsrcinfo -lelf' 'RANLIB=ranlib' 'libs' 'netlib' 'shared'
[...]
"/opt/FJSVxtclanga/tcsds-1.2.35/bin/../include/arm_sve.h", line 6: catastrophic error: #error directive: "SVE instruction support not enabled"                                                                                              
  #error "SVE instruction support not enabled"                                                                                                                                                                                              
   ^                                                                                                                                                                                                                                        
```
The problem is that OpenBLAS injects `-march=armv8-a`, but when using that 
```console
$ fcc -dM -E - < /dev/null | grep __ARM_FEATURE_SVE
#define __ARM_FEATURE_SVE 1
$ fcc -march=armv8-a -dM -E - < /dev/null | grep __ARM_FEATURE_SVE
$
```
and `arm_sve.h` throws the error above if `__ARM_FEATURE_SVE` is undefined.

**Comment by giordano** [comment]
And just to make things even more confusing, the macro _is_ defined when using the Clang mode:
```console
$ fcc -Nclang -march=armv8-a -dM -E - < /dev/null | grep __ARM_FEATURE_SVE
#define __ARM_FEATURE_SVE 1
```
Sigh.

**Comment by martin-frbg** [comment]
Thanks. Unfortunately I have never had access to Fugaku, and I am not sure if the contributor of the initial SVE enhancements tried building with the Fujitsu compiler at all. I _suspect_ the build may run better with TARGET=ARMV8SVE (which will add -march=armv8-a+sve unconditionally for what the build system thinks is gcc in Makefile.arm64). Basically what happens here is that the compiler version check "for gcc 11 or newer" fails as fcc did not like the version directive at all (which is also what causes all that noise in the build log).
Alternatively, you could kill the ifdef (and corresponding endif) in Makefile.arm64 lines 230/237

**Comment by martin-frbg** [comment]
As an aside, could you please provide the output of `fcc -dM -E ctest.c` ? Getting fcc recognized with its own options section in Makefile.arm64 would probably be the cleanest solution.

**Comment by giordano** [comment]
I had success building `develop` branch with the Fujitsu compiler in Clang mode:
```console
make -j 40 CC='fcc -Nclang' FC='frt -Nclang' INTERFACE64=1 SYMBOLSUFFIX=64_ LDFLAGS='-lfj90i -lfj90f -lfjsrcinfo -lelf' 'RANLIB=ranlib' 'libs' 'netlib' 'shared'
```
This also sets `-march=armv8.2-a+sve -mtune=a64fx` correctly, for example in the log it's written:
```
 fcc -Nclang -DCBLAS -c -O2 -DMAX_STACK_ALLOC=2048 -Wall -DF_INTERFACE_FUJITSU  -fPIC -DSMP_SERVER -DNO_WARMUP -DMAX_CPU_NUMBER=60 -DMAX_PARALLEL_NUMBER=1 -DBUILD_SINGLE=1 -DBUILD_DOUBLE=1 -DBUILD_COMPLEX=1 -DBUILD_COMPLEX16=1 -DVERSION=\"0.3.20\" -march=armv8.2-a+sve -mtune=a64fx -UASMNAME -UASMFNAME -UNAME -UCNAME -UCHAR_NAME -UCHAR_CNAME -DASMNAME=cblas_dspr2 -DASMFNAME=cblas_dspr2_ -DNAME=cblas_dspr2_ -DCNAME=cblas_dspr2 -DCHAR_NAME=\"cblas_dspr2_\" -DCHAR_CNAME=\"cblas_dspr2\" -DNO_AFFINITY -I.. -I. -DDOUBLE  -UCOMPLEX spr2.c -o cblas_dspr2.o
```

---
Regarding the tests:
```console
$ fcc -dM -E - < ctest.c
#define __CHAR_BIT__ 8
#define __PRAGMA_REDEFINE_EXTNAME 1
#define __STDC__ 1
#define __STDC_VERSION__ 201112L
#define __STDC_NO_ATOMICS__ 1
#define __STDC_UTF_16__ 1
#define __STDC_UTF_32__ 1
#define __STDC_HOSTED__ 1
#define __EDG__ 1
#define __EDG_VERSION__ 500
#define __EDG_SIZE_TYPE__ unsigned long
#define __EDG_PTRDIFF_TYPE__ long
#define __GNUC_STDC_INLINE__ 1
#define __GNUC__ 6
#define __GNUC_MINOR__ 1
#define __GNUC_PATCHLEVEL__ 0
#define __VERSION__ "EDG gcc 6.1 mode"
#define __CHAR16_TYPE__ unsigned short
#define __CHAR32_TYPE__ unsigned int
#define __ATOMIC_RELAXED 0
#define __ATOMIC_CONSUME 1
#define __ATOMIC_ACQUIRE 2
#define __ATOMIC_RELEASE 3
#define __ATOMIC_ACQ_REL 4
#define __ATOMIC_SEQ_CST 5
#define __GCC_ATOMIC_BOOL_LOCK_FREE 2
#define __GCC_ATOMIC_CHAR_LOCK_FREE 2
#define __GCC_ATOMIC_CHAR16_T_LOCK_FREE 2
#define __GCC_ATOMIC_CHAR32_T_LOCK_FREE 2
#define __GCC_ATOMIC_WCHAR_T_LOCK_FREE 2
#define __GCC_ATOMIC_SHORT_LOCK_FREE 2
#define __GCC_ATOMIC_INT_LOCK_FREE 2
#define __GCC_ATOMIC_LONG_LOCK_FREE 2
#define __GCC_ATOMIC_LLONG_LOCK_FREE 2
#define __GCC_ATOMIC_POINTER_LOCK_FREE 2
#define __STDC_NO_THREADS__ 1
#define __unix__ 1
#define __PTRDIFF_TYPE__ long int
#define __SIZE_TYPE__ long unsigned int
#define __WCHAR_TYPE__ int
#define __linux__ 1
#define unix 1
#define linux 1
#define __FUJITSU 1
#define __FCC_major__ 4
#define __FCC_minor__ 8
#define __FCC_patchlevel__ 0
#define __FCC_version__ "4.8.0"
#define __aarch64__ 1
#define __unix 1
#define __fcc_version__ 0x800
#define __fcc_version 800
#define __USER_LABEL_PREFIX__ 
#define __OPTIMIZE__ 1
#define __ARM_ARCH 8
#define __ARM_FEATURE_SVE 1
#define __FP_FAST_FMA 1
#define __ELF__ 1
#define __linux 1
#define __LIBC_6B 1
#define __LP64__ 1
#define _LP64 1
#define __uint128_t long double
#define __builtin_assume_aligned(address,align) (address)
$ fcc -Nclang -dM -E - < ctest.c
#define _LP64 1
#define __AARCH64EL__ 1
#define __ARM_64BIT_STATE 1
#define __ARM_ACLE 200
#define __ARM_ALIGN_MAX_STACK_PWR 4
#define __ARM_ARCH 8
#define __ARM_ARCH_ISA_A64 1
#define __ARM_ARCH_PROFILE 'A'
#define __ARM_FEATURE_CLZ 1
#define __ARM_FEATURE_CRC32 1
#define __ARM_FEATURE_CRYPTO 1
#define __ARM_FEATURE_DIRECTED_ROUNDING 1
#define __ARM_FEATURE_DIV 1
#define __ARM_FEATURE_FMA 1
#define __ARM_FEATURE_FP16_SCALAR_ARITHMETIC 1
#define __ARM_FEATURE_FP16_VECTOR_ARITHMETIC 1
#define __ARM_FEATURE_IDIV 1
#define __ARM_FEATURE_LDREX 0xF
#define __ARM_FEATURE_NUMERIC_MAXMIN 1
#define __ARM_FEATURE_QRDMX 1
#define __ARM_FEATURE_SVE 1
#define __ARM_FEATURE_UNALIGNED 1
#define __ARM_FP 0xE
#define __ARM_FP16_ARGS 1
#define __ARM_FP16_FORMAT_IEEE 1
#define __ARM_NEON 1
#define __ARM_NEON_FP 0xE
#define __ARM_PCS_AAPCS64 1
#define __ARM_SIZEOF_MINIMAL_ENUM 4
#define __ARM_SIZEOF_WCHAR_T 4
#define __ATOMIC_ACQUIRE 2
#define __ATOMIC_ACQ_REL 4
#define __ATOMIC_CONSUME 1
#define __ATOMIC_RELAXED 0
#define __ATOMIC_RELEASE 3
#define __ATOMIC_SEQ_CST 5
#define __BIGGEST_ALIGNMENT__ 16
#define __BYTE_ORDER__ __ORDER_LITTLE_ENDIAN__
#define __CHAR16_TYPE__ unsigned short
#define __CHAR32_TYPE__ unsigned int
#define __CHAR_BIT__ 8
#define __CHAR_UNSIGNED__ 1
#define __CLANG_ATOMIC_BOOL_LOCK_FREE 2
#define __CLANG_ATOMIC_CHAR16_T_LOCK_FREE 2
#define __CLANG_ATOMIC_CHAR32_T_LOCK_FREE 2
#define __CLANG_ATOMIC_CHAR_LOCK_FREE 2
#define __CLANG_ATOMIC_INT_LOCK_FREE 2
#define __CLANG_ATOMIC_LLONG_LOCK_FREE 2
#define __CLANG_ATOMIC_LONG_LOCK_FREE 2
#define __CLANG_ATOMIC_POINTER_LOCK_FREE 2
#define __CLANG_ATOMIC_SHORT_LOCK_FREE 2
#define __CLANG_ATOMIC_WCHAR_T_LOCK_FREE 2
#define __CLANG_FUJITSU 1
#define __CONSTANT_CFSTRINGS__ 1
#define __DBL_DECIMAL_DIG__ 17
#define __DBL_DENORM_MIN__ 4.9406564584124654e-324
#define __DBL_DIG__ 15
#define __DBL_EPSILON__ 2.2204460492503131e-16
#define __DBL_HAS_DENORM__ 1
#define __DBL_HAS_INFINITY__ 1
#define __DBL_HAS_QUIET_NAN__ 1
#define __DBL_MANT_DIG__ 53
#define __DBL_MAX_10_EXP__ 308
#define __DBL_MAX_EXP__ 1024
#define __DBL_MAX__ 1.7976931348623157e+308
#define __DBL_MIN_10_EXP__ (-307)
#define __DBL_MIN_EXP__ (-1021)
#define __DBL_MIN__ 2.2250738585072014e-308
#define __DECIMAL_DIG__ __LDBL_DECIMAL_DIG__
#define __ELF__ 1
#define __FCC_major__ 4
#define __FCC_minor__ 8
#define __FCC_patchlevel__ 0
#define __FCC_version__ "4.8.0"
#define __FINITE_MATH_ONLY__ 0
#define __FLT16_DECIMAL_DIG__ 5
#define __FLT16_DENORM_MIN__ 5.9604644775390625e-8F16
#define __FLT16_DIG__ 3
#define __FLT16_EPSILON__ 9.765625e-4F16
#define __FLT16_HAS_DENORM__ 1
#define __FLT16_HAS_INFINITY__ 1
#define __FLT16_HAS_QUIET_NAN__ 1
#define __FLT16_MANT_DIG__ 11
#define __FLT16_MAX_10_EXP__ 4
#define __FLT16_MAX_EXP__ 15
#define __FLT16_MAX__ 6.5504e+4F16
#define __FLT16_MIN_10_EXP__ (-13)
#define __FLT16_MIN_EXP__ (-14)
#define __FLT16_MIN__ 6.103515625e-5F16
#define __FLT_DECIMAL_DIG__ 9
#define __FLT_DENORM_MIN__ 1.40129846e-45F
#define __FLT_DIG__ 6
#define __FLT_EPSILON__ 1.19209290e-7F
#define __FLT_EVAL_METHOD__ 0
#define __FLT_HAS_DENORM__ 1
#define __FLT_HAS_INFINITY__ 1
#define __FLT_HAS_QUIET_NAN__ 1
#define __FLT_MANT_DIG__ 24
#define __FLT_MAX_10_EXP__ 38
#define __FLT_MAX_EXP__ 128
#define __FLT_MAX__ 3.40282347e+38F
#define __FLT_MIN_10_EXP__ (-37)
#define __FLT_MIN_EXP__ (-125)
#define __FLT_MIN__ 1.17549435e-38F
#define __FLT_RADIX__ 2
#define __FP_FAST_FMA 1
#define __FP_FAST_FMAF 1
#define __GCC_ATOMIC_BOOL_LOCK_FREE 2
#define __GCC_ATOMIC_CHAR16_T_LOCK_FREE 2
#define __GCC_ATOMIC_CHAR32_T_LOCK_FREE 2
#define __GCC_ATOMIC_CHAR_LOCK_FREE 2
#define __GCC_ATOMIC_INT_LOCK_FREE 2
#define __GCC_ATOMIC_LLONG_LOCK_FREE 2
#define __GCC_ATOMIC_LONG_LOCK_FREE 2
#define __GCC_ATOMIC_POINTER_LOCK_FREE 2
#define __GCC_ATOMIC_SHORT_LOCK_FREE 2
#define __GCC_ATOMIC_TEST_AND_SET_TRUEVAL 1
#define __GCC_ATOMIC_WCHAR_T_LOCK_FREE 2
#define __GCC_HAVE_SYNC_COMPARE_AND_SWAP_1 1
#define __GCC_HAVE_SYNC_COMPARE_AND_SWAP_2 1
#define __GCC_HAVE_SYNC_COMPARE_AND_SWAP_4 1
#define __GCC_HAVE_SYNC_COMPARE_AND_SWAP_8 1
#define __GNUC_MINOR__ 2
#define __GNUC_PATCHLEVEL__ 1
#define __GNUC_STDC_INLINE__ 1
#define __GNUC__ 4
#define __GXX_ABI_VERSION 1002
#define __INT16_C_SUFFIX__ 
#define __INT16_FMTd__ "hd"
#define __INT16_FMTi__ "hi"
#define __INT16_MAX__ 32767
#define __INT16_TYPE__ short
#define __INT32_C_SUFFIX__ 
#define __INT32_FMTd__ "d"
#define __INT32_FMTi__ "i"
#define __INT32_MAX__ 2147483647
#define __INT32_TYPE__ int
#define __INT64_C_SUFFIX__ L
#define __INT64_FMTd__ "ld"
#define __INT64_FMTi__ "li"
#define __INT64_MAX__ 9223372036854775807L
#define __INT64_TYPE__ long int
#define __INT8_C_SUFFIX__ 
#define __INT8_FMTd__ "hhd"
#define __INT8_FMTi__ "hhi"
#define __INT8_MAX__ 127
#define __INT8_TYPE__ signed char
#define __INTMAX_C_SUFFIX__ L
#define __INTMAX_FMTd__ "ld"
#define __INTMAX_FMTi__ "li"
#define __INTMAX_MAX__ 9223372036854775807L
#define __INTMAX_TYPE__ long int
#define __INTMAX_WIDTH__ 64
#define __INTPTR_FMTd__ "ld"
#define __INTPTR_FMTi__ "li"
#define __INTPTR_MAX__ 9223372036854775807L
#define __INTPTR_TYPE__ long int
#define __INTPTR_WIDTH__ 64
#define __INT_FAST16_FMTd__ "hd"
#define __INT_FAST16_FMTi__ "hi"
#define __INT_FAST16_MAX__ 32767
#define __INT_FAST16_TYPE__ short
#define __INT_FAST32_FMTd__ "d"
#define __INT_FAST32_FMTi__ "i"
#define __INT_FAST32_MAX__ 2147483647
#define __INT_FAST32_TYPE__ int
#define __INT_FAST64_FMTd__ "ld"
#define __INT_FAST64_FMTi__ "li"
#define __INT_FAST64_MAX__ 9223372036854775807L
#define __INT_FAST64_TYPE__ long int
#define __INT_FAST8_FMTd__ "hhd"
#define __INT_FAST8_FMTi__ "hhi"
#define __INT_FAST8_MAX__ 127
#define __INT_FAST8_TYPE__ signed char
#define __INT_LEAST16_FMTd__ "hd"
#define __INT_LEAST16_FMTi__ "hi"
#define __INT_LEAST16_MAX__ 32767
#define __INT_LEAST16_TYPE__ short
#define __INT_LEAST32_FMTd__ "d"
#define __INT_LEAST32_FMTi__ "i"
#define __INT_LEAST32_MAX__ 2147483647
#define __INT_LEAST32_TYPE__ int
#define __INT_LEAST64_FMTd__ "ld"
#define __INT_LEAST64_FMTi__ "li"
#define __INT_LEAST64_MAX__ 9223372036854775807L
#define __INT_LEAST64_TYPE__ long int
#define __INT_LEAST8_FMTd__ "hhd"
#define __INT_LEAST8_FMTi__ "hhi"
#define __INT_LEAST8_MAX__ 127
#define __INT_LEAST8_TYPE__ signed char
#define __INT_MAX__ 2147483647
#define __LDBL_DECIMAL_DIG__ 36
#define __LDBL_DENORM_MIN__ 6.47517511943802511092443895822764655e-4966L
#define __LDBL_DIG__ 33
#define __LDBL_EPSILON__ 1.92592994438723585305597794258492732e-34L
#define __LDBL_HAS_DENORM__ 1
#define __LDBL_HAS_INFINITY__ 1
#define __LDBL_HAS_QUIET_NAN__ 1
#define __LDBL_MANT_DIG__ 113
#define __LDBL_MAX_10_EXP__ 4932
#define __LDBL_MAX_EXP__ 16384
#define __LDBL_MAX__ 1.18973149535723176508575932662800702e+4932L
#define __LDBL_MIN_10_EXP__ (-4931)
#define __LDBL_MIN_EXP__ (-16381)
#define __LDBL_MIN__ 3.36210314311209350626267781732175260e-4932L
#define __LITTLE_ENDIAN__ 1
#define __LONG_LONG_MAX__ 9223372036854775807LL
#define __LONG_MAX__ 9223372036854775807L
#define __LP64__ 1
#define __OBJC_BOOL_IS_BOOL 0
#define __OPENCL_MEMORY_SCOPE_ALL_SVM_DEVICES 3
#define __OPENCL_MEMORY_SCOPE_DEVICE 2
#define __OPENCL_MEMORY_SCOPE_SUB_GROUP 4
#define __OPENCL_MEMORY_SCOPE_WORK_GROUP 1
#define __OPENCL_MEMORY_SCOPE_WORK_ITEM 0
#define __OPTIMIZE__ 1
#define __ORDER_BIG_ENDIAN__ 4321
#define __ORDER_LITTLE_ENDIAN__ 1234
#define __ORDER_PDP_ENDIAN__ 3412
#define __POINTER_WIDTH__ 64
#define __PRAGMA_REDEFINE_EXTNAME 1
#define __PTRDIFF_FMTd__ "ld"
#define __PTRDIFF_FMTi__ "li"
#define __PTRDIFF_MAX__ 9223372036854775807L
#define __PTRDIFF_TYPE__ long int
#define __PTRDIFF_WIDTH__ 64
#define __SCHAR_MAX__ 127
#define __SHRT_MAX__ 32767
#define __SIG_ATOMIC_MAX__ 2147483647
#define __SIG_ATOMIC_WIDTH__ 32
#define __SIZEOF_DOUBLE__ 8
#define __SIZEOF_FLOAT__ 4
#define __SIZEOF_INT128__ 16
#define __SIZEOF_INT__ 4
#define __SIZEOF_LONG_DOUBLE__ 16
#define __SIZEOF_LONG_LONG__ 8
#define __SIZEOF_LONG__ 8
#define __SIZEOF_POINTER__ 8
#define __SIZEOF_PTRDIFF_T__ 8
#define __SIZEOF_SHORT__ 2
#define __SIZEOF_SIZE_T__ 8
#define __SIZEOF_WCHAR_T__ 4
#define __SIZEOF_WINT_T__ 4
#define __SIZE_FMTX__ "lX"
#define __SIZE_FMTo__ "lo"
#define __SIZE_FMTu__ "lu"
#define __SIZE_FMTx__ "lx"
#define __SIZE_MAX__ 18446744073709551615UL
#define __SIZE_TYPE__ long unsigned int
#define __SIZE_WIDTH__ 64
#define __STDC_HOSTED__ 1
#define __STDC_UTF_16__ 1
#define __STDC_UTF_32__ 1
#define __STDC_VERSION__ 201112L
#define __STDC__ 1
#define __UINT16_C_SUFFIX__ 
#define __UINT16_FMTX__ "hX"
#define __UINT16_FMTo__ "ho"
#define __UINT16_FMTu__ "hu"
#define __UINT16_FMTx__ "hx"
#define __UINT16_MAX__ 65535
#define __UINT16_TYPE__ unsigned short
#define __UINT32_C_SUFFIX__ U
#define __UINT32_FMTX__ "X"
#define __UINT32_FMTo__ "o"
#define __UINT32_FMTu__ "u"
#define __UINT32_FMTx__ "x"
#define __UINT32_MAX__ 4294967295U
#define __UINT32_TYPE__ unsigned int
#define __UINT64_C_SUFFIX__ UL
#define __UINT64_FMTX__ "lX"
#define __UINT64_FMTo__ "lo"
#define __UINT64_FMTu__ "lu"
#define __UINT64_FMTx__ "lx"
#define __UINT64_MAX__ 18446744073709551615UL
#define __UINT64_TYPE__ long unsigned int
#define __UINT8_C_SUFFIX__ 
#define __UINT8_FMTX__ "hhX"
#define __UINT8_FMTo__ "hho"
#define __UINT8_FMTu__ "hhu"
#define __UINT8_FMTx__ "hhx"
#define __UINT8_MAX__ 255
#define __UINT8_TYPE__ unsigned char
#define __UINTMAX_C_SUFFIX__ UL
#define __UINTMAX_FMTX__ "lX"
#define __UINTMAX_FMTo__ "lo"
#define __UINTMAX_FMTu__ "lu"
#define __UINTMAX_FMTx__ "lx"
#define __UINTMAX_MAX__ 18446744073709551615UL
#define __UINTMAX_TYPE__ long unsigned int
#define __UINTMAX_WIDTH__ 64
#define __UINTPTR_FMTX__ "lX"
#define __UINTPTR_FMTo__ "lo"
#define __UINTPTR_FMTu__ "lu"
#define __UINTPTR_FMTx__ "lx"
#define __UINTPTR_MAX__ 18446744073709551615UL
#define __UINTPTR_TYPE__ long unsigned int
#define __UINTPTR_WIDTH__ 64
#define __UINT_FAST16_FMTX__ "hX"
#define __UINT_FAST16_FMTo__ "ho"
#define __UINT_FAST16_FMTu__ "hu"
#define __UINT_FAST16_FMTx__ "hx"
#define __UINT_FAST16_MAX__ 65535
#define __UINT_FAST16_TYPE__ unsigned short
#define __UINT_FAST32_FMTX__ "X"
#define __UINT_FAST32_FMTo__ "o"
#define __UINT_FAST32_FMTu__ "u"
#define __UINT_FAST32_FMTx__ "x"
#define __UINT_FAST32_MAX__ 4294967295U
#define __UINT_FAST32_TYPE__ unsigned int
#define __UINT_FAST64_FMTX__ "lX"
#define __UINT_FAST64_FMTo__ "lo"
#define __UINT_FAST64_FMTu__ "lu"
#define __UINT_FAST64_FMTx__ "lx"
#define __UINT_FAST64_MAX__ 18446744073709551615UL
#define __UINT_FAST64_TYPE__ long unsigned int
#define __UINT_FAST8_FMTX__ "hhX"
#define __UINT_FAST8_FMTo__ "hho"
#define __UINT_FAST8_FMTu__ "hhu"
#define __UINT_FAST8_FMTx__ "hhx"
#define __UINT_FAST8_MAX__ 255
#define __UINT_FAST8_TYPE__ unsigned char
#define __UINT_LEAST16_FMTX__ "hX"
#define __UINT_LEAST16_FMTo__ "ho"
#define __UINT_LEAST16_FMTu__ "hu"
#define __UINT_LEAST16_FMTx__ "hx"
#define __UINT_LEAST16_MAX__ 65535
#define __UINT_LEAST16_TYPE__ unsigned short
#define __UINT_LEAST32_FMTX__ "X"
#define __UINT_LEAST32_FMTo__ "o"
#define __UINT_LEAST32_FMTu__ "u"
#define __UINT_LEAST32_FMTx__ "x"
#define __UINT_LEAST32_MAX__ 4294967295U
#define __UINT_LEAST32_TYPE__ unsigned int
#define __UINT_LEAST64_FMTX__ "lX"
#define __UINT_LEAST64_FMTo__ "lo"
#define __UINT_LEAST64_FMTu__ "lu"
#define __UINT_LEAST64_FMTx__ "lx"
#define __UINT_LEAST64_MAX__ 18446744073709551615UL
#define __UINT_LEAST64_TYPE__ long unsigned int
#define __UINT_LEAST8_FMTX__ "hhX"
#define __UINT_LEAST8_FMTo__ "hho"
#define __UINT_LEAST8_FMTu__ "hhu"
#define __UINT_LEAST8_FMTx__ "hhx"
#define __UINT_LEAST8_MAX__ 255
#define __UINT_LEAST8_TYPE__ unsigned char
#define __USER_LABEL_PREFIX__ 
#define __VERSION__ "4.2.1 Compatible Fujitsu Clang 7.1.0"
#define __WCHAR_MAX__ 4294967295U
#define __WCHAR_TYPE__ unsigned int
#define __WCHAR_UNSIGNED__ 1
#define __WCHAR_WIDTH__ 32
#define __WINT_MAX__ 4294967295U
#define __WINT_TYPE__ unsigned int
#define __WINT_UNSIGNED__ 1
#define __WINT_WIDTH__ 32
#define __aarch64__ 1
#define __clang__ 1
#define __clang_major__ 7
#define __clang_minor__ 1
#define __clang_patchlevel__ 0
#define __clang_version__ "7.1.0 "
#define __gnu_linux__ 1
#define __linux 1
#define __linux__ 1
#define __llvm__ 1
#define __unix 1
#define __unix__ 1
#define linux 1
#define unix 1
```

**Comment by martin-frbg** [comment]
Thank you - should be fixed by #3699 now
